client/state/themes/utils.js

```diff
@@ -190,6 +191,22 @@ export function getSerializedThemesQueryWithoutPage( query, siteId ) {
 }
 
 /**
+ * Filter wpcom themes from Jetpack theme list
+ * Themes on Jetpack installed from WordPress.com have -wpcom suffix.
+ * We filter out all those themes because they will also be visible on
+ * second list specific to WordPress.com. This may be too simple aproach
+ * so it may revisit it again later.
+ *
+ * TODO Veriy that this aproach is sufficien.
+ *
+ * @param  {Array}  themes Array of theme objects
+ * @return {Array}         Filtered themes
+ */
+export function filterWpcomThemesFromJetpack( themes ) {
+	return filter( themes, theme => ! endsWith( theme.id, '-wpcom' ) );
+}
```

Comment:
Naming is tricky because we do not know which version we will use more often, this one or `isWpcomTheme` selector when we will make it for other purposes. I will still name it `isWpcomTheme` because it makes sense :)